internal/operands/common-templates/reconcile.go

```diff
@@ -122,6 +127,52 @@ func reconcileEditRole(request *common.Request) (common.ResourceStatus, error) {
 			foundRole.Rules = newRole.Rules
 		})
 }
+func updateOldTemplates(request *common.Request) (common.ResourceStatus, error) {
+	baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during preparing of type label requirement, %v", err))
+		panic(err)
+	}
+	var versionRequirement *labels.Requirement
+	versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{Version})
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during preparing of version label requirement, %v", err))
+		panic(err)
+	}
+
+	labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+
+	opts := client.ListOptions{
+		LabelSelector: labelsSelector,
+	}
+
+	var oldTemplates templatev1.TemplateList
+	err = request.Client.List(request.Context, &oldTemplates, &opts)
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during listing templates, %v", err))
+		return common.ResourceStatus{}, err
+	}
+
+	for _, template := range oldTemplates.Items {
+		needsUpdate := false
+		newTemplate := template.DeepCopy()
```

Comment:
I don't think the DeepCopy and MergeFrom are necessary.
You could just store a list of labels that need to be removed and then go over them and build the json patch directly.
Would be a few less/smaller allocations than DeepCopy.
But it should be okay this way.

Reply:
`DeepCopy` is redundant here, since you intend to `Patch` anyway